### ISSUE TITLE
Editing and saving `Directory.Build.props` causes `AggregateException`

### ISSUE THREAD

**Issue by just-ero** [microsoft/vscode-dotnettools#22]
Type: <b>Bug</b>

Edited my solution's root `Directory.Build.props` in the new solution view by removing the following entries, as they are currently in preview and do not work in VSCode:
```xml
<UseArtifactsOutput>true</UseArtifactsOutput>
<ArtifactsPath>$(MSBuildThisFileDirectory)artifacts</ArtifactsPath>
```

This resulted in the following stacktrace:
```
=====================
6/6/2023 11:38:37 PM
LimitedFunctionality
System.AggregateException: Project system data flow 'Source: 3664974' closed because of an exception.
missingBuildCount >= 0 (missingBuildCount >= 0)
---> (Inner Exception #0) Microsoft.Assumes+InternalErrorException: missingBuildCount >= 0
   at Microsoft.Assumes.Fail(String message)
   at Microsoft.Assumes.True(Boolean condition, String message)
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencies.MSBuildDependencyCollection.<>c__DisplayClass5_0.<TryUpdate>g__PopulateMissingBuildData|5(<>c__DisplayClass5_1&)
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencies.MSBuildDependencyCollection.<>c__DisplayClass5_0.<TryUpdate>g__ComposeItemUpdates|2()
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencies.MSBuildDependencyCollection.<>c__DisplayClass5_0.<TryUpdate>g__ProcessAddsAndUpdates|0()
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencies.MSBuildDependencyCollection.TryUpdate(IProjectChangeDescription evaluationProjectChange, IProjectChangeDescription buildProjectChange, String projectFullPath, Nullable`1& dependencies)
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencySubscriber.Source.<LinkExternalInput>g__Transform|6_1(IProjectSubscriptionUpdate update)
   at Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.Derive[T,TOut](IProjectVersionedValue`1 projectVersionedValue, Func`2 transform)
   at Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Subscriptions.MSBuildDependencySubscriber.Source.<LinkExternalInput>b__6_0(IProjectVersionedValue`1 u)
   at Microsoft.VisualStudio.ProjectSystem.TransformBlockSlim`2.TransformBlockSlimSync.ProcessInputAsync(TInput input)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
===================
```

Extension version: 0.1.83
VS Code version: Code 1.78.2 (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:39:26.248Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 5800X 8-Core Processor              (16 x 4200)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|63.91GB (54.92GB free)|
|Process Argv|C:\\Users\\Ero\\Desktop --crash-reporter-id eecc004c-9bbb-4437-b727-31efcf441836|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
282f8724:30602487
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
7ij38806:30736111
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30730052
hidesbindicator:30760978
pythongtdpath:30739704

```

</details>

**Comment by kvenkatrajan** [comment]
Thanks for your feedback, we have triaged the issue and are prepping a fix for this.

**Comment by smitpatel** [comment]
This issue has been fixed and will be available in next release.

**Comment by smitpatel** [comment]
The fix is available in v0.2.99
